llvm/lib/MC/MCDwarf.cpp

```diff
@@ -243,17 +256,36 @@ void MCDwarfLineTable::emitOne(
     Discriminator = 0;
     LastLine = LineEntry.getLine();
     LastLabel = Label;
+    IsAtStartSeq = false;
   }
 
   // Generate DWARF line end entry.
   // We do not need this for DwarfDebug that explicitly terminates the line
   // table using ranges whenever CU or section changes. However, the MC path
   // does not track ranges nor terminate the line table. In that case,
   // conservatively use the section end symbol to end the line table.
-  if (!EndEntryEmitted)
+  if (!EndEntryEmitted && !IsAtStartSeq)
```

Comment:
Missing test coverage: the test doesn't fail if I remove `&& !IsAtStartSeq`

Reply:
Added coverage.